src/plots/mapbox/layout_attributes.js

```diff
@@ -100,7 +100,7 @@ var attrs = module.exports = overrideAll({
         },
         sourcetype: {
             valType: 'enumerated',
-            values: ['geojson', 'vector'],
+            values: ['geojson', 'vector', 'raster'],
```

Comment:
At some point, we could add `'image'` https://docs.mapbox.com/mapbox-gl-js/style-spec/#sources-image

which could be used together with `mapbox.layers[i].type: 'raster'`